types/block.go

```diff
@@ -638,11 +638,8 @@ type hasher struct {
 func (h hasher) Hash() []byte {
 	hasher := tmhash.New()
 	if h.item != nil && !cmn.IsTypedNil(h.item) && !cmn.IsEmpty(h.item) {
-		bz, err := cdc.MarshalBinaryBare(h.item)
-		if err != nil {
-			panic(err)
-		}
-		_, err = hasher.Write(bz)
+		bz := cdc.MustMarshalBinaryBare(h.item)
+		_, err := hasher.Write(bz)
```

Comment:
I know this didn't exist before, but can you add a comment that we currently use sha256 which can't err and hence it is OK to ignore the error here?